### ISSUE TITLE
Endless character output from the keyboard 

### ISSUE THREAD

**Issue by Mike-0815** [joncampbell123/dosbox-x#4838]
### Describe the bug

Hello Jon,

I am using DOSBOX-X SDL 1, latest version (DOSBOX-X vsbuild v20231007) under Windows 7 x64 Pro. DOSBOX-X works mostly fine for me, but there is an annoying error:

When you quit some programs and return to DOS, DOSBOX-X will put out the last character pressed endlessly until you press any button on keyboard.

That behaviour does not occur on DOSBOX, DOSBOX ECE or PCem.

Also I used DOSBOX-X without any frontend, to know, it's not a bug in conjuncion with the frontend.

Thanks for attencion.

Kind regards,
Mike

### Steps to reproduce the behaviour

Just download Phils DOS Benchmarkpack from PhilsComputerLab:
https://www.philscomputerlab.com/dos-benchmark-pack.html
and start any of the Quake Benchmarks there and when you gonna quit Quake, you have to press "y" (I have to press "z" because it's a german keyboard). Then, when quitting to DOS, you will see the letter repeatedly outputted, until you press any key.

### Expected behavior

Of course I expect, that after returning to DOS, the last pressed key will not be outputted, nor be seen in the DOS-Shell.

### What operating system(s) this bug have occurred on?

Windos 7 x64 Pro

### What version(s) of DOSBox-X have this bug?

vsbuild v20231007 x64 SDL 1 -- (SDL 2 build I didn't try)

### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by joncampbell123** [comment]
Can you try output=surface?

There are platforms where whenever the SDL library is told to resize that keyboard status can get lost during mode switches.

**Comment by Mike-0815** [comment]
I tried output=surface, but no change at all, same problem. One think I have to tell as well: When I press "z" (german keyboard) in Quake to exit, the first keystroke will be somehow ignored, means Quake will not quit (seems to do nothing), but the second keystroke then will exit Quake.

**Comment by maron2000** [comment]
Windows VS SDL2 version is working fine with German keyboard.
No problem whether SDL1/2 with Japanese keyboard.
Video output is Direct3D (I think it is the default)

**Comment by Mike-0815** [comment]
Yes, I can confirm that. But it would be very nice, if SDL1 version works the same without that error.

**Comment by maron2000** [comment]
SDL1 is already obsolete in terms of adding new features, so don't expect much from it.
SDL2 should be quite fine for a request of cosmetic stuff.
